pulsar-broker/src/main/java/org/apache/pulsar/broker/namespace/OwnershipCache.java

```diff
@@ -258,7 +261,12 @@ public OwnedBundle getOwnedBundle(NamespaceBundle bundle) {
         CompletableFuture<OwnedBundle> future = ownedBundlesCache.getIfPresent(bundle);
 
         if (future != null && future.isDone() && !future.isCompletedExceptionally()) {
-            return future.join();
+            try {
+                return future.get(pulsar.getConfiguration().getZooKeeperOperationTimeoutSeconds(), TimeUnit.SECONDS);
+            } catch (Exception e) {
+                log.error("Get owned bundle failed ", e);
+                return null;
```

Comment:
This is a behaviour change.
We should fail and not return null

Reply:
Nice view, done